src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -38,21 +46,54 @@ protected GameManagementService(RoomRepository repository, SocketService socketS
         currentGameMap = new HashMap<>();
     }
 
-    public Game createAddGameFromRoom(Room room) {
-        // TODO: Create the game from the room (or roomId).
-        Game game = new Game();
-        currentGameMap.put(room.getRoomId(), game);
-        return null;
-    }
-
     protected Game getGameFromRoomId(String roomId) {
-        return currentGameMap.get(roomId);
+        Game game = currentGameMap.get(roomId);
+        if (game == null) {
+            throw new ApiException(GameError.NOT_FOUND);
+        }
+
+        return game;
     }
 
     protected void removeGame(String roomId) {
         currentGameMap.remove(roomId);
     }
 
+    public GameDto getGameDto(String roomId) {
+        return GameMapper.toDto(getGameFromRoomId(roomId));
+    }
+
+    // When host starts the game, redirect everyone and initialize the game state
+    public RoomDto startGame(String roomId, StartGameRequest request) {
+        Room room = repository.findRoomByRoomId(roomId);
+
+        // If requested room does not exist in database, throw an exception.
+        if (room == null) {
+            throw new ApiException(RoomError.NOT_FOUND);
+        }
+
+        // If user making request is not the host, throw an exception.
+        if (!request.getInitiator().getNickname().equals(room.getHost().getNickname())) {
+            throw new ApiException(RoomError.INVALID_PERMISSIONS);
+        }
+
+        room.setActive(true);
+        repository.save(room);
+
+        // Initialize game state
+        createAddGameFromRoom(room);
```

Comment:
Eventually, there will be other "conversion" steps like making the timer start and any necessary steps for taking care of room settings; however, no other changes now. This looks great!